data/option_test.go

```diff
@@ -0,0 +1,65 @@
+package data_test
+
+import (
+	"GoSlice/data"
+	"GoSlice/util/test"
+	"strings"
+	"testing"
+)
+
+func TestFanSpeedDefaults(t *testing.T) {
+	opts := data.DefaultOptions()
+	test.Equals(t, opts.Filament.FanSpeed.LayerToSpeedLUT[2], 255)
+}
+
+func TestFanSpeedNegative(t *testing.T) {
+	fanSpeedOptions := data.FanSpeedOptions{}
+	err := fanSpeedOptions.Set("1=-5")
+	test.Assert(t, strings.Contains(err.Error(), "fan control needs to be in format"), "fan speed error expected")
+}
+
+func TestFanSpeedLayerNegative(t *testing.T) {
+	fanSpeedOptions := data.FanSpeedOptions{}
+	err := fanSpeedOptions.Set("-1=5")
+	test.Assert(t, strings.Contains(err.Error(), "fan control needs to be in format"), "fan speed error expected")
+}
+
+func TestFanSpeedGreaterThan255(t *testing.T) {
+	fanSpeedOptions := data.FanSpeedOptions{}
+	err := fanSpeedOptions.Set("1=256")
+	test.Assert(t, strings.Contains(err.Error(), "fan control needs to be in format"), "fan speed error expected")
+}
+
+func TestFanSpeedSingleGood(t *testing.T) {
+	fanSpeedOptions := data.FanSpeedOptions{}
+	err := fanSpeedOptions.Set("1=20")
+	test.Assert(t, err == nil, "unsuccessfully set single fan speed")
+}
+
+func TestFanSpeedMultipleGood(t *testing.T) {
+	fanSpeedOptions := data.FanSpeedOptions{}
+	err := fanSpeedOptions.Set("1=20,5=100")
+	test.Assert(t, err == nil, "unsuccessfully set multiple fan speed")
+}
+
+func TestFanSpeedMultipleOneBadOneGood(t *testing.T) {
+	fanSpeedOptions := data.FanSpeedOptions{}
+	err := fanSpeedOptions.Set("1=-20,5=100")
+	test.Assert(t, strings.Contains(err.Error(), "fan control needs to be in format"), "fan speed error expected")
+}
+
+func TestFanSpeedStringMultipleGood(t *testing.T) {
+	fanSpeedText := "1=20,5=100"
+	fanSpeedOptions := data.FanSpeedOptions{}
+	err := fanSpeedOptions.Set(fanSpeedText)
+	test.Assert(t, err == nil, "unsuccessfully set multiple fan speed")
+	test.Equals(t, fanSpeedOptions.String(), fanSpeedText)
```

Comment:
A test should not test two functions at the same time if possible. That's why I would not use also the .String here and instead write extra tests for it.

Instead you can use a custom comperator as I wrote in the table-test-example above. then you can just write:
`test.Equals(t, fanSpeedOptions.String(), fanSpeedText, fanSpeedOptionsComparer())`

To test the .String just add another test function using the table driven test pattern to test some cases for it.

Reply:
done. Although was unsure if its really worth it for 3 tests, but yes, might grow eventually :)